tests/unit/errors/conftest.py

```diff
@@ -0,0 +1,84 @@
+import json
+from collections import namedtuple
+
+import pytest
+import requests
+
+_TestResponse = namedtuple("_TestResponse", ("data", "r", "method", "url"))
+
+
+def _mk_response(
+    data, status, method=None, url=None, headers=None, data_transform=None
+):
+    resp = requests.Response()
+
+    if data_transform:
+        resp._content = data_transform(data).encode("utf-8")
+    else:
+        resp._content = data.encode("utf-8")
+    resp.encoding = "utf-8"
+
+    if headers:
+        resp.headers.update(headers)
+
+    resp.status_code = str(status)
+    resp.reason = {
+        200: "OK",
+        400: "Bad Request",
+        401: "Unauthorized",
+        403: "Forbidden",
+        404: "Not Found",
+        500: "Server Error",
+    }.get(status, "Unknown")
+    method = method or "GET"
+    url = url or "default-example-url.bogus"
+    resp.url = url
+    resp.request = requests.Request(method=method, url=url, headers=headers)
```

Comment:
Should the response headers be added to the headers of the request, too? This might be convenient behavior but I want to confirm whether `headers` should be split into customizable `response_headers` and `request_headers`.

Reply:
I'll take a look. This was, as you may have guessed, largely a lift-and-shift.
I think maybe `headers=...` and `request_headers`, since this whole fixture is supposed to be a response builder.

Reply:
If something depends on this existing functionality, no need to pursue this at all, but I wanted to ask in case this is a good time opportunity to separate the two.